Client-side remove package from hand when put on chain

KptKosmit91 opened the pull request:
Forces mainhand package to be instantly removed from the mainhand when placing it on a chain conveyor to prevent any potential duping

This fixes (for survival mode atleast) a weird compatibility issue with Epicfight, where `InputEvents.onClickInput` is executed twice, which dupes the package. This seems to be a bigger issue in itself and i don't think that's in the scope of this pull request. I have instead posted more about that issue on the Create Discord in this message: https://discord.com/channels/620934202875183104/689866656914210897/1379140295165476945

KptKosmit91 committed:
Client-side remove package from hand when put on chain

Forces mainhand package to be instantly removed from the mainhand when placing it on a chain conveyor

This is more of a failsafe to prevent the player from putting the package on the chain multiple times

VoidLeech commented:
Relevant issues #8508 (package dupe when clicking multiple times quickly (here due to #6901))
Neither issue is fully fixed by this, it's kinda a bandaid. (We shouldn't be getting to this code a second time to begin with)

KptKosmit91 committed:
Mech arm optimizations

-Lazy updating during SEARCH_INPUTS phase so as to not make the arm search for targets on every tick. This should help with performance with many arms or arms that have lots of targets
-Replaced if-else-if blocks that checked the arm's Phase with Switch statements
-Cached getSlotCount result for arm interaction point slot iterations, should help when iterating high slot counts

VoidLeech commented:
Superceded by #8608 for the earlier commit
![image](https://github.com/user-attachments/assets/7533aa8a-a719-4947-9a6a-8d5491a50807)

and #8613 for the unrelated latter.
